File: backend/services/email.js
// Banglasketch – Professional Email Notification Service
const nodemailer = require("nodemailer");
const { parseUserAgent, formatDateBST } = require("./adminCredentials");

// ---------------------------------------------------------------------------
// Configuration & Helpers
// ---------------------------------------------------------------------------
const BRAND = {
  name: "Banglasketch",
  bn: "বাংলা স্কিচ",
  tagline: "Luxury Interior Architecture & Design Studio",
  colors: {
    gold: "#C5A059",
    goldLight: "#E5C778",
    goldDark: "#936D28",
    navy: "#0A2540",
    navyDeep: "#061A30",
    red: "#E07B2A",
    white: "#FAF7F2",
    muted: "#9CA3AF",
    success: "#10B981",
  },
  contact: process.env.CONTACT_EMAIL || "info@banglasketch.com",
  phone: "01712-458794",
};

function escapeHtml(value) {
  return String(value ?? "")
    .replace(/&/g, "&amp;")
    .replace(/</g, "&lt;")
    .replace(/>/g, "&gt;")
    .replace(/"/g, "&quot;")
    .replace(/'/g, "&#39;");
}

function maskEmail(email) {
  if (!email) return "";
  const [local, domain] = email.toLowerCase().split("@");
  if (!domain) return email;
  const visible = Math.min(2, Math.max(1, local.length - 1));
  const masked =
    local.slice(0, visible) +
    "*".repeat(Math.max(0, local.length - visible - 1)) +
    local.slice(-1);
  return `${masked}@${domain}`;
}

function getTransporter() {
  const user = (process.env.SMTP_USER || "").trim();
  const pass = (process.env.SMTP_PASS || "").replace(/\s+/g, "");
  const port = Number(process.env.SMTP_PORT) || 587;
  const secure = process.env.SMTP_SECURE === "true" || port === 465;

  return nodemailer.createTransport({
    host: process.env.SMTP_HOST || "smtp.gmail.com",
    port,
    secure,
    auth: { user, pass },
    connectionTimeout: 10000,
    greetingTimeout: 5000,
    socketTimeout: 10000,
  });
}

function isPlaceholderCredential(value) {
  if (!value) return true;
  const v = String(value).trim().toLowerCase();
  return (
    v === "your_app_password" ||
    v === "your_email@gmail.com" ||
    v === "your_email" ||
    v === "password"
  );
}

function canSendMail() {
  const user = (process.env.SMTP_USER || "").trim();
  const pass = (process.env.SMTP_PASS || "").replace(/\s+/g, "");
  return Boolean(
    user &&
      pass &&
      !isPlaceholderCredential(user) &&
      !isPlaceholderCredential(pass)
  );
}

// ---------------------------------------------------------------------------
// Email Templates (Luxury Brand System)
// ---------------------------------------------------------------------------

const createEmailContainer = (heading, subHeading, children) => `
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>${escapeHtml(heading)}</title>
</head>
<body style="margin:0;padding:0;background-color:#F4EFE6;font-family:-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Helvetica,Arial,sans-serif;-webkit-font-smoothing:antialiased;">
  <table role="presentation" width="100%" cellpadding="0" cellspacing="0" border="0" style="background-color:#F4EFE6;padding:40px 12px;">
    <tr>
      <td align="center">
        <!-- Main Card Container -->
        <table role="presentation" width="100%" cellpadding="0" cellspacing="0" border="0" style="max-width:600px;background:#FFFFFF;border-radius:24px;overflow:hidden;box-shadow:0 25px 60px rgba(10,37,64,0.14);border:1px solid #EBE4D8;">
          <!-- Luxury Header -->
          <tr>
            <td align="center" style="background:linear-gradient(135deg, ${BRAND.colors.navy} 0%, ${BRAND.colors.navyDeep} 100%);padding:45px 35px;text-align:center;">
              <table role="presentation" cellpadding="0" cellspacing="0" border="0" align="center">
                <tr>
                  <td align="center">
                    <span style="display:inline-block;padding:4px 14px;background:rgba(197,160,89,0.18);border:1px solid rgba(229,199,120,0.35);border-radius:20px;color:${BRAND.colors.goldLight};font-size:10px;font-weight:600;letter-spacing:3px;text-transform:uppercase;margin-bottom:12px;">
                      ${BRAND.bn} • ARCHITECTURAL STUDIO
                    </span>
                    <h1 style="color:#FAF7F2;margin:8px 0 0;font-size:30px;font-weight:700;letter-spacing:2px;font-family:Georgia,'Times New Roman',serif;">
                      ${BRAND.name}
                    </h1>
                    <p style="color:${BRAND.colors.muted};margin:8px 0 0;font-size:12px;letter-spacing:3px;text-transform:uppercase;">
                      ${BRAND.tagline}
                    </p>
                    <div style="width:50px;height:2px;background:linear-gradient(90deg, transparent, ${BRAND.colors.gold}, transparent);margin:20px auto 0;"></div>
                  </td>
                </tr>
              </table>
            </td>
          </tr>

          <!-- Content Body -->
          <tr>
            <td style="padding:42px 38px;background:#FFFFFF;">
              <h2 style="color:${BRAND.colors.navy};margin:0 0 10px;font-size:23px;font-weight:700;letter-spacing:-0.5px;font-family:Georgia,'Times New Roman',serif;">
                ${escapeHtml(heading)}
              </h2>
              <p style="color:#4B5563;margin:0 0 28px;font-size:15px;line-height:1.65;">
                ${escapeHtml(subHeading)}
              </p>
              ${children}
            </td>
          </tr>

          <!-- Luxury Footer -->
          <tr>
            <td style="background:#F9F8F5;padding:32px 36px;text-align:center;border-top:1px solid #EFECE6;">
              <p style="margin:0 0 8px;font-size:13px;color:#6B7280;">
                <strong style="color:${BRAND.colors.navy};font-weight:700;">${BRAND.name}</strong> • ${BRAND.bn}
              </p>
              <p style="margin:0 0 14px;font-size:12px;color:#9CA3AF;line-height:1.5;">
                Hashem Mansion, Level-1, 48 Kazi Nazrul Islam Ave, Dhaka 1215<br>
                Direct Concierge: +${BRAND.phone} • <a href="mailto:${BRAND.contact}" style="color:${BRAND.colors.goldDark};text-decoration:none;">${BRAND.contact}</a>
              </p>
              <div style="width:30px;height:1px;background:#D1D5DB;margin:12px auto;"></div>
              <p style="margin:0;font-size:11px;color:#9CA3AF;">
                &copy; ${new Date().getFullYear()} ${BRAND.name}. Confidential Security Notice.
              </p>
            </td>
          </tr>
        </table>
      </td>
    </tr>
  </table>
</body>
</html>`;

// ---------------------------------------------------------------------------
// Senders
// ---------------------------------------------------------------------------

const sendContactNotification = async (submission) => {
  if (!canSendMail()) return { skipped: true };

  const name = escapeHtml(submission.name);
  const email = escapeHtml(submission.email);
  const phone = escapeHtml(submission.phone || "—");
  const service = escapeHtml(submission.service_type || "—");
  const message = escapeHtml(submission.message);
  const subjectName = String(submission.name || "Website visitor").replace(/[\r\n]/g, " ");

  const html = createEmailContainer(
    "New Client Inquiry Received",
    "A prospective client has submitted a design request via the Banglasketch website.",
    `
    <table style="width:100%;border-collapse:separate;border-spacing:0 12px;font-size:14px;">
      <tr>
        <td style="padding:14px 18px;background:${BRAND.colors.navy};color:${BRAND.colors.goldLight};border-radius:12px 0 0 12px;font-weight:600;">Name</td>
        <td style="padding:14px 18px;background:#FFFFFF;color:#1F2937;border-radius:0 12px 12px 0;font-weight:500;">${name}</td>
      </tr>
      <tr>
        <td style="padding:14px 18px;background:${BRAND.colors.navy};color:${BRAND.colors.goldLight};border-radius:12px 0 0 12px;font-weight:600;">Email</td>
        <td style="padding:14px 18px;background:#FFFFFF;color:#1F2937;border-radius:0 12px 12px 0;"><a href="mailto:${email}" style="color:${BRAND.colors.red};text-decoration:none;">${email}</a></td>
      </tr>
      <tr>
        <td style="padding:14px 18px;background:${BRAND.colors.navy};color:${BRAND.colors.goldLight};border-radius:12px 0 0 12px;font-weight:600;">Phone</td>
        <td style="padding:14px 18px;background:#FFFFFF;color:#1F2937;border-radius:0 12px 12px 0;">${phone}</td>
      </tr>
      <tr>
        <td style="padding:14px 18px;background:${BRAND.colors.navy};color:${BRAND.colors.goldLight};border-radius:12px 0 0 12px;font-weight:600;">Service</td>
        <td style="padding:14px 18px;background:#FFFFFF;color:#1F2937;border-radius:0 12px 12px 0;font-weight:500;">${service}</td>
      </tr>
      <tr>
        <td style="padding:14px 18px;background:${BRAND.colors.navy};color:${BRAND.colors.goldLight};border-radius:12px 0 0 12px;font-weight:600;vertical-align:top;">Message</td>
        <td style="padding:14px 18px;background:#FFFFFF;color:#1F2937;border-radius:0 12px 12px 0;line-height:1.6;">${message}</td>
      </tr>
    </table>
    <p style="text-align:center;margin-top:24px;color:${BRAND.colors.muted};font-size:11px;">Notification received at ${new Date().toLocaleString()}</p>
  `
  );

  await getTransporter().sendMail({
    from: `"${BRAND.name} Inquiries" <${process.env.SMTP_USER}>`,
    to: BRAND.contact,
    subject: `New Client Inquiry — ${subjectName}`,
    html,
  });
};

const sendClientConfirmation = async (submission) => {
  if (!canSendMail()) return { skipped: true };

  const name = escapeHtml(submission.name);
  const message = escapeHtml(submission.message);

  const html = createEmailContainer(
    `Thank You, ${name}!`,
    `We have received your inquiry and our senior design team will review it shortly. We aim to respond within 24 hours to discuss how we can transform your space.`,
    `
    <div style="background:#FFFFFF;border-left:4px solid ${BRAND.colors.gold};padding:20px;border-radius:8px;margin:24px 0;">
      <p style="color:${BRAND.colors.goldDark};font-size:12px;font-weight:700;text-transform:uppercase;letter-spacing:1px;margin:0 0 10px;">Your Inquiry Summary</p>
      <p style="color:#374151;margin:0;font-style:italic;line-height:1.7;">"${message}"</p>
    </div>
    <p style="color:#6B7280;font-size:14px;line-height:1.6;">In the meantime, feel free to reach us directly:</p>
    <p style="color:${BRAND.colors.navy};font-size:15px;font-weight:600;margin:8px 0;">📞 ${BRAND.phone}</p>
    <p style="color:#6B7280;font-size:14px;margin:0;">🌐 <a href="/" style="color:${BRAND.colors.red};text-decoration:none;">Visit banglasketch.com</a></p>
  `
  );

  await getTransporter().sendMail({
    from: `"${BRAND.name}" <${process.env.SMTP_USER}>`,
    to: submission.email,
    subject: `We received your message — ${BRAND.name}`,
    html,
  });
};

// ---------------------------------------------------------------------------
// Admin Password Reset & Security Notifications
// ---------------------------------------------------------------------------

function buildOtpEmailHtml(otp, targetEmail, expiresInMinutes = 10) {
  const masked = maskEmail(targetEmail);

  const html = createEmailContainer(
    "Password Reset Verification Code",
    `A master password reset request was initiated for your Banglasketch CMS account associated with <strong style="color:${BRAND.colors.navy};">${masked}</strong>. Use the single-use security code below to proceed:`,
    `
    <!-- OTP Code Display Card -->
    <table role="presentation" width="100%" cellpadding="0" cellspacing="0" border="0" style="margin:28px 0;">
      <tr>
        <td align="center">
          <div style="background:#FBF8F3;border:2px dashed ${BRAND.colors.gold};border-radius:18px;padding:34px 24px;text-align:center;max-width:380px;box-shadow:0 8px 30px rgba(197,160,89,0.12);">
            <div style="font-size:11px;letter-spacing:4px;text-transform:uppercase;color:${BRAND.colors.goldDark};margin-bottom:14px;font-weight:700;">
              One-Time Verification Code
            </div>
            <div style="font-size:44px;letter-spacing:12px;font-weight:800;color:${BRAND.colors.navy};font-family:'Courier New',Courier,monospace;line-height:1;padding-left:12px;">
              ${otp}
            </div>
            <p style="margin:16px 0 0;font-size:12px;color:#6B7280;">
              Enter this code on the password reset screen.<br>Never share this code with anyone.
            </p>
          </div>
        </td>
      </tr>
    </table>

    <!-- Security Meta Badges (Table Layout for Maximum Client Compatibility) -->
    <table role="presentation" width="100%" cellpadding="0" cellspacing="0" border="0" style="margin:28px 0 10px;border-top:1px solid #F3F4F6;padding-top:20px;">
      <tr>
        <td align="center" width="33%" style="padding:10px 5px;vertical-align:top;">
          <div style="width:38px;height:38px;border-radius:50%;background:#0A2540;color:#E5C778;line-height:38px;text-align:center;margin:0 auto 8px;font-size:16px;">⏱</div>
          <p style="margin:0;font-size:12px;color:#4B5563;font-weight:500;">Expires in</p>
          <p style="margin:2px 0 0;font-size:12px;color:${BRAND.colors.red};font-weight:700;">${expiresInMinutes} Minutes</p>
        </td>
        <td align="center" width="33%" style="padding:10px 5px;vertical-align:top;">
          <div style="width:38px;height:38px;border-radius:50%;background:#D1FAE5;color:#059669;line-height:38px;text-align:center;margin:0 auto 8px;font-size:16px;">🔒</div>
          <p style="margin:0;font-size:12px;color:#4B5563;font-weight:500;">Encrypted</p>
          <p style="margin:2px 0 0;font-size:12px;color:#059669;font-weight:700;">End-to-End</p>
        </td>
        <td align="center" width="33%" style="padding:10px 5px;vertical-align:top;">
          <div style="width:38px;height:38px;border-radius:50%;background:#FEE2E2;color:#DC2626;line-height:38px;text-align:center;margin:0 auto 8px;font-size:16px;">⚠️</div>
          <p style="margin:0;font-size:12px;color:#4B5563;font-weight:500;">Didn't request?</p>
          <p style="margin:2px 0 0;font-size:12px;color:#DC2626;font-weight:700;">Ignore safely</p>
        </td>
      </tr>
    </table>
  `
  );
  return html;
}

const sendPasswordResetOTP = async ({ to, otp, expiresInMinutes = 10 }) => {
  const target = (to || process.env.ADMIN_RECOVERY_EMAIL || "arifulhaquerohan@gmail.com").trim().toLowerCase();
  const html = buildOtpEmailHtml(otp, target, expiresInMinutes);

  if (canSendMail()) {
    try {
      await getTransporter().sendMail({
        from: `"${BRAND.name} Security" <${process.env.SMTP_USER}>`,
        to: target,
        subject: "Banglasketch CMS Password Reset Code",
        html,
      });
      console.log(`✓ Admin password reset OTP successfully sent to ${target}`);
      return { sent: true, devMode: false };
    } catch (error) {
      console.error("⚠ Failed to send admin OTP email via SMTP:", error.message);
      return { sent: false, devMode: true, error: error.message };
    }
  }
  return { sent: false, error: "Mail delivery unavailable" };
};

const sendAdminLoginNotification = async ({ to, ip, userAgent, timestamp }) => {
  const target = (to || process.env.ADMIN_NOTIFICATION_EMAIL || process.env.ADMIN_RECOVERY_EMAIL || process.env.SMTP_USER || "arifulhaquerohan@gmail.com").trim().toLowerCase();
  const clientIp = escapeHtml(ip || "Unknown IP");
  const device = escapeHtml(parseUserAgent(userAgent));
  const { bstStr, utcStr } = formatDateBST(timestamp);

  const html = createEmailContainer(
    "Security Alert: Admin Login Detected",
    "A successful login to the Banglasketch CMS Administration Panel was just detected.",
    `
    <div style="background:#FFFFFF;border-radius:12px;padding:20px;margin:20px 0;border:1px solid #E5E7EB;">
      <table style="width:100%;border-collapse:collapse;font-size:13px;color:#374151;">
        <tr>
          <td style="padding:10px 0;color:${BRAND.colors.navy};font-weight:700;width:140px;border-bottom:1px solid #F3F4F6;">Date & Time (BST)</td>
          <td style="padding:10px 0;border-bottom:1px solid #F3F4F6;font-family:monospace;">${bstStr}</td>
        </tr>
        <tr>
          <td style="padding:10px 0;color:${BRAND.colors.navy};font-weight:700;border-bottom:1px solid #F3F4F6;">UTC Timestamp</td>
          <td style="padding:10px 0;border-bottom:1px solid #F3F4F6;font-size:12px;color:#6B7280;">${utcStr}</td>
        </tr>
        <tr>
          <td style="padding:10px 0;color:${BRAND.colors.navy};font-weight:700;border-bottom:1px solid #F3F4F6;">IP Address</td>
          <td style="padding:10px 0;border-bottom:1px solid #F3F4F6;font-family:monospace;color:${BRAND.colors.red};">${clientIp}</td>
        </tr>
        <tr>
          <td style="padding:10px 0;color:${BRAND.colors.navy};font-weight:700;border-bottom:1px solid #F3F4F6;">Device & Browser</td>
          <td style="padding:10px 0;border-bottom:1px solid #F3F4F6;">${device}</td>
        </tr>
        <tr>
          <td style="padding:10px 0;color:${BRAND.colors.navy};font-weight:700;">Status</td>
          <td style="padding:10px 0;">
            <span style="background:rgba(16,185,129,0.15);color:#059669;border:1px solid rgba(16,185,129,0.3);padding:4px 12px;border-radius:20px;font-size:12px;font-weight:600;">
              ✓ Authorized Session
            </span>
          </td>
        </tr>
      </table>
    </div>
    <div style="background:#FEF2F2;border:1px solid #FECACA;border-radius:10px;padding:16px;margin-top:20px;">
      <p style="color:#DC2626;margin:0 0 6px;font-size:13px;font-weight:700;">⚠️ Didn't initiate this login?</p>
      <p style="color:#6B7280;margin:0;font-size:12px;line-height:1.5;">If this activity was not you, please use the <strong>Forgot Password</strong> option immediately to revoke access and secure your account.</p>
    </div>
  `
  );

  if (canSendMail()) {
    try {
      await getTransporter().sendMail({
        from: `"${BRAND.name} Security" <${process.env.SMTP_USER}>`,
        to: target,
        subject: `Security Alert: Admin Login Detected — Bangla Sketch`,
        html,
      });
      console.log(`✓ Admin login notification email sent to ${target}`);
      return { sent: true };
    } catch (error) {
      console.error("⚠ Failed to send admin login notification email:", error.message);
      return { sent: false, error: error.message };
    }
  }
  console.log(`[DEV MODE] Admin login alert for ${target} from ${clientIp} (${device})`);
  return { sent: false, devMode: true };
};

const sendPasswordResetSuccessNotification = async ({ to, ip, timestamp }) => {
  const target = (to || process.env.ADMIN_RECOVERY_EMAIL || process.env.SMTP_USER || "arifulhaquerohan@gmail.com").trim().toLowerCase();
  const clientIp = escapeHtml(ip || "Unknown IP");
  const { bstStr, utcStr } = formatDateBST(timestamp);

  const html = createEmailContainer(
    "Password Reset Successful",
    "Your Banglasketch CMS administrator master password was successfully changed.",
    `
    <div style="text-align:center;margin:24px 0;">
      <div style="width:64px;height:64px;border-radius:50%;background:#D1FAE5;color:#059669;display:flex;align-items:center;justify-content:center;margin:0 auto 16px;font-size:28px;">✓</div>
      <p style="color:#059669;font-weight:600;font-size:16px;">Access Credentials Updated</p>
    </div>
    <div style="background:#FFFFFF;border-radius:12px;padding:20px;margin:20px 0;border:1px solid #E5E7EB;">
      <table style="width:100%;border-collapse:collapse;font-size:13px;color:#374151;">
        <tr>
          <td style="padding:10px 0;color:${BRAND.colors.navy};font-weight:700;width:140px;border-bottom:1px solid #F3F4F6;">Reset Time (BST)</td>
          <td style="padding:10px 0;border-bottom:1px solid #F3F4F6;font-family:monospace;">${bstStr}</td>
        </tr>
        <tr>
          <td style="padding:10px 0;color:${BRAND.colors.navy};font-weight:700;border-bottom:1px solid #F3F4F6;">UTC Timestamp</td>
          <td style="padding:10px 0;border-bottom:1px solid #F3F4F6;font-size:12px;color:#6B7280;">${utcStr}</td>
        </tr>
        <tr>
          <td style="padding:10px 0;color:${BRAND.colors.navy};font-weight:700;">Origin IP</td>
          <td style="padding:10px 0;border-bottom:1px solid #F3F4F6;font-family:monospace;color:${BRAND.colors.red};">${clientIp}</td>
        </tr>
      </table>
    </div>
    <p style="color:#6B7280;font-size:13px;line-height:1.6;margin-top:20px;">
      If you initiated this change, no further action is needed. If you did not make this change, please recover your account immediately.
    </p>
  `
  );

  if (canSendMail()) {
    try {
      await getTransporter().sendMail({
        from: `"${BRAND.name} Security" <${process.env.SMTP_USER}>`,
        to: target,
        subject: `Banglasketch CMS Password Reset Successful`,
        html,
      });
      console.log(`✓ Admin password reset confirmation email sent to ${target}`);
      return { sent: true };
    } catch (error) {
      console.error("⚠ Failed to send password reset confirmation email:", error.message);
      return { sent: false, error: error.message };
    }
  }
  return { sent: false, devMode: true };
};

module.exports = {
  canSendMail,
  sendContactNotification,
  sendClientConfirmation,
  sendPasswordResetOTP,
  sendAdminLoginNotification,
  sendPasswordResetSuccessNotification,
  parseUserAgent,
  maskEmail,
  escapeHtml,
  BRAND,
};

File: django_backend/apps/leads/services/email.py
import logging
import datetime
from django.conf import settings
from django.core.mail import EmailMultiAlternatives
from django.utils import timezone
from django.utils.html import escape
from apps.leads.models import ContactEmailJob, ContactSubmission

logger = logging.getLogger(__name__)

def build_notification_email(submission: ContactSubmission) -> tuple[str, str, str]:
    """Build subject, plain text, and HTML body for studio admin notification."""
    subject = f"[Banglasketch Inquiry] {submission.name} - {submission.service_type or 'General Inquiry'}"

    text_content = (
        f"New Contact Form Submission\n"
        f"============================\n"
        f"Name: {submission.name}\n"
        f"Email: {submission.email}\n"
        f"Phone: {submission.phone or 'Not provided'}\n"
        f"Service: {submission.service_type or 'General'}\n"
        f"Submitted At: {submission.submitted_at}\n\n"
        f"Message:\n"
        f"{submission.message}\n"
    )

    html_content = f"""
    <!DOCTYPE html>
    <html>
    <head>
      <meta charset="utf-8">
      <style>
        body {{ font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif; background-color: #f7f6f2; margin: 0; padding: 24px; color: #242824; }}
        .card {{ background: #ffffff; border-radius: 12px; padding: 32px; max-width: 600px; margin: 0 auto; border: 1px solid #e2ded7; }}
        .header {{ border-bottom: 2px solid #586348; padding-bottom: 16px; margin-bottom: 24px; }}
        .title {{ font-size: 20px; font-weight: 700; color: #242824; margin: 0; }}
        .field {{ margin-bottom: 12px; }}
        .label {{ font-size: 12px; font-weight: 600; text-transform: uppercase; color: #586348; letter-spacing: 0.5px; }}
        .value {{ font-size: 15px; color: #242824; margin-top: 2px; }}
        .message-box {{ background: #fcfaef; border-left: 4px solid #586348; padding: 16px; margin-top: 16px; border-radius: 4px; font-size: 14px; line-height: 1.6; white-space: pre-wrap; }}
        .footer {{ margin-top: 32px; font-size: 12px; color: #8c948c; text-align: center; }}
      </style>
    </head>
    <body>
      <div class="card">
        <div class="header">
          <h1 class="title">Banglasketch Studio — New Inquiry</h1>
        </div>
        <div class="field">
          <div class="label">Client Name</div>
          <div class="value">{escape(submission.name)}</div>
        </div>
        <div class="field">
          <div class="label">Email Address</div>
          <div class="value"><a href="mailto:{escape(submission.email)}">{escape(submission.email)}</a></div>
        </div>
        <div class="field">
          <div class="label">Phone Number</div>
          <div class="value">{escape(submission.phone or 'Not provided')}</div>
        </div>
        <div class="field">
          <div class="label">Space / Service</div>
          <div class="value">{escape(submission.service_type or 'General Inquiry')}</div>
        </div>
        <div class="field">
          <div class="label">Message</div>
          <div class="message-box">{escape(submission.message)}</div>
        </div>
        <div class="footer">
          Received via Banglasketch Web Portal on {timezone.localtime(submission.submitted_at).strftime('%B %d, %Y at %I:%M %p')}
        </div>
      </div>
    </body>
    </html>
    """
    return subject, text_content, html_content


def build_confirmation_email(submission: ContactSubmission) -> tuple[str, str, str]:
    """Build subject, plain text, and HTML body for client confirmation."""
    subject = "Thank you for contacting Banglasketch Studio"

    text_content = (
        f"Dear {submission.name},\n\n"
        f"Thank you for reaching out to Banglasketch Studio. We have received your message regarding {submission.service_type or 'our architectural services'}.\n\n"
        f"Our team is reviewing your inquiry and will respond within 1-2 business days.\n\n"
        f"Warm regards,\n"
        f"Banglasketch Studio\n"
        f"Architecture • Interior • Landscape\n"
        f"Dhaka, Bangladesh\n"
    )

    html_content = f"""
    <!DOCTYPE html>
    <html>
    <head>
      <meta charset="utf-8">
      <style>
        body {{ font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif; background-color: #f7f6f2; margin: 0; padding: 24px; color: #242824; }}
        .card {{ background: #ffffff; border-radius: 12px; padding: 36px; max-width: 600px; margin: 0 auto; border: 1px solid #e2ded7; }}
        .brand {{ font-size: 22px; font-weight: 800; letter-spacing: 1px; color: #242824; text-transform: uppercase; margin-bottom: 8px; }}
        .subtitle {{ font-size: 13px; text-transform: uppercase; letter-spacing: 2px; color: #586348; margin-bottom: 24px; }}
        .content {{ font-size: 15px; line-height: 1.7; color: #3d443d; margin-bottom: 24px; }}
        .summary {{ background: #fbf9f4; border: 1px solid #ede8de; border-radius: 8px; padding: 16px; margin: 20px 0; font-size: 14px; }}
        .footer {{ border-top: 1px solid #eae5db; padding-top: 20px; font-size: 12px; color: #8c948c; line-height: 1.5; }}
      </style>
    </head>
    <body>
      <div class="card">
        <div class="brand">Banglasketch</div>
        <div class="subtitle">Architectural Design Studio</div>
        <div class="content">
          <p>Dear {escape(submission.name)},</p>
          <p>Thank you for reaching out to us. We have received your inquiry and our team is currently reviewing your project details.</p>
          <p>We strive to bring thoughtful architectural and interior craftsmanship to every space, and we look forward to exploring how we can collaborate on your vision.</p>
          <div class="summary">
            <strong>Requested Service:</strong> {escape(submission.service_type or 'General Inquiry')}<br>
            <strong>Reference ID:</strong> #{submission.id}
          </div>
          <p>A member of our team will be in touch with you shortly.</p>
        </div>
        <div class="footer">
          <strong>Banglasketch Studio</strong><br>
          Dhaka, Bangladesh • info@banglasketch.com<br>
          <a href="https://banglasketch.com" style="color: #586348; text-decoration: none;">banglasketch.com</a>
        </div>
      </div>
    </body>
    </html>
    """
    return subject, text_content, html_content


def process_email_job(job: ContactEmailJob) -> bool:
    """
    Processes a single ContactEmailJob.
    Returns True if sent successfully, False otherwise.
    """
    submission = job.submission
    from_email = getattr(settings, "DEFAULT_FROM_EMAIL", "info@banglasketch.com")

    try:
        if job.kind == "notification":
            admin_email = getattr(settings, "ADMIN_RECOVERY_EMAIL", "arifulhaquerohan@gmail.com")
            subject, text, html = build_notification_email(submission)
            msg = EmailMultiAlternatives(subject, text, from_email, [admin_email])
            msg.attach_alternative(html, "text/html")
            msg.send(fail_silently=False)
        elif job.kind == "confirmation":
            if not submission.email or "@" not in submission.email:
                # Can't deliver confirmation if no valid email
                job.delivered_at = timezone.now()
                job.save(update_fields=["delivered_at"])
                return True
            subject, text, html = build_confirmation_email(submission)
            msg = EmailMultiAlternatives(subject, text, from_email, [submission.email])
            msg.attach_alternative(html, "text/html")
            msg.send(fail_silently=False)
        else:
            logger.warning(f"Unknown email job kind: {job.kind}")
            return False

        job.delivered_at = timezone.now()
        job.attempts += 1
        job.save(update_fields=["delivered_at", "attempts"])
        return True

    except Exception as e:
        logger.error(f"Failed to deliver email job #{job.id} ({job.kind}): {e}")
        job.attempts += 1
        # Exponential backoff: 10s * 2^(attempts-1), max 1 hour
        backoff_seconds = min(3600, 10 * (2 ** max(0, job.attempts - 1)))
        job.available_at = timezone.now() + datetime.timedelta(seconds=backoff_seconds)
        job.save(update_fields=["attempts", "available_at"])
        return False
